### ISSUE TITLE
[Wait for CLC-2105] Unit address is not saved after changing country

### ISSUE THREAD

**Issue by tobi-or-not-tobi** [SAP/spartacus#8907]
After changing the country for an address assigned to US (with a specific region), saving throws an error in the console. It's unclear what is happening, it seems that the backend is not dropping the region, while moving to a new country. This might happen as we're patching the address, with handing over a new country, and an empty region. 

```
{
   "errors" : [ {
      "message" : "No region with the code US-HI found.",
      "type" : "ConversionError"
   } ]
}
```

**NOTE**
This is actually caused by a bug in the backend: https://cxjira.sap.com/browse/CLC-2105

**Comment by Platonn** [comment]
I confirm the bug. I created address in United States and region Louisiana and saved. Then I tried editing the country and reseting the region, but I got `400` error from backend when I've tried saving changes. I tried sending to backend various alternatives for `region`:
- no `region` field
- `region: null`
- `region: { isocode: null }`

In each case I got the `400` error complaining that the Louisiana region cannot be found. 

```
{
   "errors" : [ {
      "message" : "No region with the code US-LA found.",
      "type" : "ConversionError"
   } ]
}
```

---
**Question** to the backend team: Is there any other way I can reset the region for the country?

**Comment by Platonn** [comment]
Some alternative possible workaounds:
1. delete and re-create address, when the country has been changed
   - pros: transparent to user
   - cons: more complexity (custom case to handle) with different http calls, 
   - unknowns: deleting and re-creating will change the address ID. Will it be a problem? User would expect he only edited the data of old address.

2. disable the `Country` field specifically in the edit form of the unit-address. 
   - pros: complexity encapsulated only in the UI, not in the facade logic or http calls
   - cons: user may be confused why he is not allowed to change the address Country. If he needs to change country, he could workaround it by manually removing and re-creating the address though.

**Comment by Platonn** [comment]
I can confirm that the same issue happens when editing Country in the standard `my-account/address-book` section, even in B2C storefront (http://localhost:4200/electronics-spa/en/USD/my-account/address-book)
HTTP PATCH `/occ/v2/electronics-spa/users/current/addresses/:addressId`
```
country: {isocode: "TR"}  // changed from Unated States to Turkey
isocode: "TR"
defaultAddress: true
firstName: "asd"
lastName: "asd"
line1: "asdasd"
line2: "asd"
phone: "1243"
postalCode: "asd"
titleCode: "mr"
town: "asd"
```
Got 400 error:
```
{
   "errors" : [ {
      "message" : "No region with the code US-MA found.",
      "type" : "ConversionError"
   } ]
}
```
FYI @frankgrupe

**Comment by Platonn** [comment]
btw. I observed other issues when editing unit address (unrelated to this ticket):
- When created US address, refreshed page and the opening the edit from, the Regions control is loaded with a delay (due to penging backend call for regions - which is OK), but when loaded, the original value `region.isocode` is reset to `null`, but should remain the original value. The issue doesn't happen when regions for US has been loaded previously during the same session and are available synchronously.
- When succesfuly edited the address, we get the error in the constructor of the `UpdateAddressSuccess` ngrx action. It's because a wrong object is passed to the `UpdateAddressSuccess` action in the `OrgUnitsEffects`. We pass `data` (which is `null` by design - see [docs](https://spartacus-dev0.eastus.cloudapp.azure.com:9002/occ/v2/swagger-ui.html#/Organizational%20Unit%20Management/updateOrgUnit)), but we should pass the `address` object instead. See source code:
https://github.com/SAP/spartacus/blob/736066b/feature-libs/organization/administration/core/store/effects/org-unit.effect.ts#L403-L408
- after successfull edit of address I get infinte http loads for endpoint: `occ/v2/powertools-spa/users/current/orgUnits/:unitId`

**Comment by Pucek9** [comment]
Some issues (infinite calls, empty address success payload) fixed by Address data is not refreshed after saving + looped unit data calls after address update #8908

Created new issue related to Empty object request while create/update address #9572

**Comment by Pucek9** [comment]
I think we are not abble fix on frontend that case when we update address with region to country without region.
My proposal for temporary fix is to:
- Add personalized notification: "We can't update this address. Please create a new one instead."

OR

- Disable address edit button, or just country field if there are some regions. (I don't like this solution)

**Comment by Pucek9** [comment]
With reference to [this](https://github.com/SAP/spartacus/pull/9598#pullrequestreview-521654885) comment, we need some mechanism for handle error responses in local messages. 
Depends on: #9791

**Comment by Platonn** [comment]
As discussed offline with @tobi-or-not-tobi, it's an edge case user changes country in his address. So let's postpone it until:
- backend bug is fixed https://cxjira.sap.com/browse/CLC-2105 OR
- we have a better general approach to handling errors https://github.com/SAP/spartacus/issues/7349

Note: The same problem exists also in address-bug, so removing MyCompany-related milestone

**Comment by giancorderoortiz** [comment]
Still an issue. See: 
![Screen Shot 2021-06-08 at 4 45 13 PM](https://user-images.githubusercontent.com/46171897/121255107-0840e200-c879-11eb-9508-dcc241549d3d.png)

**Comment by giancorderoortiz** [comment]
Still on hold? Can it be fixed for 4.0?
